compiler/msgs.nim

```diff
@@ -476,6 +476,7 @@ type
   TFileInfo* = object
     fullPath: string           # This is a canonical full filesystem path
     projPath*: string          # This is relative to the project's root
+    path*: string              # The original path when it was first encountered
```

Comment:
Ugh no, not yet another path thing in here, patch ``shortName``  instead perhaps.

Reply:
This is getting more complicated now I have thought about it.  If the nim compiler were to treat symlinks as normal files, for one unique ``fullPath`` (canonicalized as it is now) could correspond to different modules from different symlinks.  The information inside ``TFileInfo`` does not seem to be enough.  I'll try to make a more elaborate test case with symlinks to check.

Reply:
We don't really support symlinks they are a clusterfuck.

Reply:
If only we could all run plan9.

Reply:
Completely besides the point, Windows doesn't really support symlinks anyway so your codebase will never work with one of the most widespread OSes out there, whether you like it or not. On Unix we can error on symlink'ed files since they are ambiguous for nim's purpose as you noted.

Reply:
Supporting symlinks wouldn't make nim break on any OS that lacks symlinks (only plan9 comes to mind).

We can already use nim to create symlinks on Windows, #6287.